lib/api/CDataFrameTrainBoostedTreeClassifierRunner.cc

```diff
@@ -165,6 +165,9 @@ void CDataFrameTrainBoostedTreeClassifierRunner::writeOneRow(
         std::size_t numberClasses{classValues.size()};
         m_InferenceModelMetadata.columnNames(featureImportance->columnNames());
         m_InferenceModelMetadata.classValues(classValues);
+        m_InferenceModelMetadata.predictionFieldType(m_PredictionFieldType);
+        m_InferenceModelMetadata.predictionFieldTypeResolverWriter(
+            CDataFrameTrainBoostedTreeClassifierRunner::writePredictedCategoryValueForType);
```

Comment:
I think this should use a lambda (or std::bind can also work) and then you can ditch the new `writePredictedCategoryValueForType` method altogether. Since this is owned by this object you don't need to worry about the life time of `this` so this can just be:
```
m_InferenceModelMetadata.predictionFieldTypeResolverWriter(
        [this](const std::string& categoryValue,
               core::CRapidJsonConcurrentLineWriter& writer) {
             this->writePredictedCategoryValue(categoryValue, writer);
        });
```

Reply:
Ah, passing `this` is ok in this context. @tveasey my C++ lambda skills are lacking.